MinecraftClient/Inventory/ItemTypeExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MinecraftClient.Inventory
+{
+    public static class ItemTypeExtensions
+    {
+        /**
+         * I see "this" in the MaterialExtensions class method, why need that?
+         * public static bool IsSolid(this Material m)
```

Comment:
That's an [extension method](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods), which is an add-on for the Material enum. In that specific case where the subject of the method is not the current class, the first argument specifies what type of object is being manipulated, and explicitly reference it as a variable. so if you define `public static bool IsFood(this ItemType m)` in a different class than `ItemType`, you can still call it as a method from ItemType:
```C#
ItemType myItemType = ItemType.Apple;
if (myItemType.IsFood())
{
    //...
}
```

If you omit `this`, the method is not considered as an extension method, and you need to call it explicitly:

```C#
ItemType myItemType = ItemType.Apple;
if (ItemTypeExtensions.IsFood(myItemType))
{
    //...
}
```

It's just for convenience when you can't add a method to the class directly. (which is the case on enums)

Reply:
Thanks for your detail explanation!